tap/source/envoy_discoverer.go

```diff
@@ -0,0 +1,112 @@
+package source
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/up9inc/mizu/shared/logger"
+)
+
+const envoyBinary = "/envoy"
+
+var numberRegex = regexp.MustCompile("[0-9]+")
+
+func discoverRelevantEnvoyPids(procfs string, clusterIps []string) ([]string, error) {
+	result := make([]string, 0)
+
+	pids, err := ioutil.ReadDir(procfs)
```

Comment:
What privileges do we need to access procfs?

Reply:
With `--privileged` container we can access all.
With non-privileged we can access only what the user of the tapper container can, and that's depend on various parameters.

At this point we can start with this, if it doesn't have permission nothing happen but an error in the log.

Overtime we may find out edge cases where we can't access it, and fix it on demand. In any case, its a common thing for monitoring containers to map `/proc` of host to container

Reply:
@nimrod-up9 I think the privileges provided through

https://github.com/up9inc/mizu/blob/4c04ae39c975845e409a074b8a9cf6663518a4da/shared/kubernetes/provider.go#L715-L721

which is added by https://github.com/up9inc/mizu/pull/418 and it's `readOnlyRootFilesystem`. That seems OK.